sdk/core/Azure.Core.TestFramework/src/ClientTestBase.cs

```diff
@@ -143,6 +144,22 @@ protected internal virtual object InstrumentOperation(Type operationType, object
             return operation;
         }
 
+        protected internal T InstrumentOperation<T>(T operation) where T : Operation =>
+            (T)InstrumentOperation(typeof(T), operation);
+
+        protected object InstrumentMockOperation(Type operationType, object operation, params IInterceptor[] interceptors)
```

Comment:
Why not just use the same name and have RecordedTestBase override this one. That way, we don't have to worry about recorded tests calling this method (to Michael's point about validation).

Reply:
I'm not sure to which method you're referring since you have two in context, but to both you and @m-nash:

1. I'm not setting out to regress any functionality with classes that already extend `ClientTestBase` and use `InstrumentOperation` already. That's why I added a new method that was specific to mocking, since after the regressing PR we can't simply use a `MockTransport` et. al. with operations.
2. The generic "overload" was for ease. Not sure why a method that actually takes a `Type` parameter was ever added, but it's more convenience - and safe, given we can specify generic type constraints - to have a generic method. `RecordedTestBase` already had it and would've used a `virtcall` to call the virtual `InstrumentOperation(Type, ...)` method, and still will. Where this method was defined doesn't change the fact it still uses a `virtcall`. The behavior will be unchanged. I simply refactored the method up the hierarchy.

Reply:
I'm suggesting that we don't add a separate InstrumentMockOperation method but instead add an implentation to the existing InstrumentOperation method. We can check to see if there are any such usages that you mention. The benefit of doing this is that it would make it impossible to call the wrong InstrumentOperation method because recorded tests will always use the one defined in RecordedTestBase.

Reply:
According to these [search results](https://cs.github.com/Azure/azure-sdk-for-net?q=InstrumentOperation+ClientTestBase), outside of these related base classes, I don't see any. Everything else seems to extend from `RecordedTestBase`, including some even more-derived classes in Azure.ResourceManager.Network.

Perhaps this could work then, and that should solve the [concern](https://github.com/Azure/azure-sdk-for-net/pull/27399#discussion_r821253497) @m-nash raised.